src/Http/Controllers/SearchController.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Rapidez\Core\Http\Controllers;
+
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Str;
+
+class SearchController
+{
+    public function __invoke(Request $request)
+    {
+        $query = Str::lower($request->q);
+        if ($searchQuery = Cache::remember('search.query.' . Str::slug($query), Carbon::now()->addHour(), function () use ($query) {
```

Comment:
This could fill up Redis really quickly, do we really need the caching here? Nothing wrong with just a MySQL lookup?